pandas/_libs/lib.pyx

```diff
@@ -2046,11 +2049,18 @@ def maybe_convert_objects(ndarray[object] objects, bint try_float=0,
             else:
                 if not seen.bool_:
                     if seen.datetime_:
-                        if not seen.numeric_:
-                            return datetimes
+                        if not seen.timedelta_:
```

Comment:
can you make this 1 level higher elif's, its very hard to follow these branches currently

Reply:
Will do. As for me - this nested "if"s are ugly even before I've changed anything. Besides it is one of the good places to make a mistake (as it happens in current case).